aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -99,6 +99,12 @@ namespace impl {
     // TODO static_assert(AllowDeprecatedTypes, "You tried to register a kernel with an unsupported input type: std::vector<T>. Please use List<T> instead.");
   };
 
+  template<class T, bool AllowDeprecatedTypes>
+  struct assert_is_valid_input_type<c10::ArrayRef<T>, AllowDeprecatedTypes>
+  : assert_is_valid_input_type<T, AllowDeprecatedTypes> {
+    static_assert(!std::is_same<T, at::Scalar>::value, "You tried to register a kernel with an unsupported input type: ArrayRef<Scalar>. Please use List<int64_t>, List<double> or Tensor instead.");
```

Comment:
None of these suggested alternatives seem particularly good

Reply:
What would you propose instead? We don't support lists of scalars afaik, so they would have to use int or double lists.

Reply:
I would tell them to file a bug and explain their use case :)